### PR TITLE
Feature/add button and form

### LINKED ISSUE #6: 3.  Form and add button
## Value Proposition
As a user, I want to add new task items so that i can manage my time each day well.
## Description
3. Userstory- Form and add button (1)
3. Userstory- Form and add button (2)
## Acception Criteria
- [x] displays a new add button on the current ongoing task list
- [x] displays a form containing several inputs, task title, date, start time and duration
- [x] displays two buttons "cancel" and "create"
## Tasks
- [x] create a new branch for this feature
- [x] add a new add button on the current ongoing task list
- [x] create a form with input fields for a _task title_, _date_, _start time_ and _duration_ in a new `Create` component
- [x] create two `Button Components` "cancel" and "create" in the `Create` component
- [x] create a function _appendTask_(_task title_, _date_, _start time_, _duration_) 
- [x] hand down the appendTask function as a prop to the `Create` component
- [x] in the `Create` component use this function and `Formdata` on form submission to append the newly created task
- [x] redireck back to homepage on submission(click create button) or on cancellation

### PR DESCRIPTION
**Pull request by Mengzhen1992**
Please review my third feature. Thanks :)

### CODE REVIEW

File: components/Task.js [hunk 1]
```diff
@@ -20,24 +25,51 @@ export default function Task({ task, handleToggleTask }) {
           />
         ))}
       </ul>
+      <AddButton
+        onMouseOver={() => setHover(!hover)}
```
Comment (CathrinTruchan) [inline components/Task.js]: This seems a bit complicated. Why don't you just use the hover effect in the styled component?

File: components/create.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import styled from "styled-components";
+import { Wrap } from "../pages";
+import { useRouter } from "next/router";
+
+export default function Create({ appendTask, setPage }) {
+  console.log("appendTask: ", appendTask);
```
Comment (CathrinTruchan) [inline components/create.js]: It's better to delete the console.logs from the code, before pushing it :)

File: components/create.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  function sendForm(event) {
+    console.log("send Form");
```
Comment (CathrinTruchan) [inline components/create.js]: Same here ;)

File: components/create.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    </FormWrapMask>
+  );
+}
+
+const FormWrapMask = styled.div`
+  display: grid;
+  grid-template-columns: 1.5rem 1fr 1.5rem;
+  grid-template-rows: 8rem auto 12rem;
+  width: 100vw;
+  min-height: 100vh;
```
Comment (CathrinTruchan) [inline components/create.js]: Why do you use min-height here instead of height?

File: components/create.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  display: flex;
+  flex-flow: column wrap;
+  gap: 0.5rem;
+  background: rgba(255, 255, 255, 0.4);
+  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.1);
+  border-radius: 8px;
+  grid-column: 2 / span 1;
+  grid-row: 2 / span 1;
+  padding: 1.5rem 1rem 0.6rem 1rem;
+`;
```
Comment (CathrinTruchan) [inline components/create.js]: Do you need both grid and flex?

**Review by CathrinTruchan — COMMENTED** [review]
Great work :) I just left a few comments.

File: components/Task.js (hunk 1 above)
Comment (Mengzhen1992) [reply components/Task.js]: @CathrinTruchan because i use 2 svg from my figma design. When the mouse passes over this button, it will turn into another svg

File: components/create.js (hunk 4 above)
Comment (Mengzhen1992) [reply components/create.js]: @CathrinTruchan if the content becomes longer than my background, using height will cause problems.

File: components/create.js (hunk 5 above)
Comment (Mengzhen1992) [reply components/create.js]: @CathrinTruchan thanks for the suggestion. My styling code looks a little complicated. i try to do some improvements

File: components/Task.js [hunk 6]
```diff
@@ -1,9 +1,14 @@
...
 import TaskItem from "./TaskItem";
+import addButton from "../public/images/addButton.svg";
+import addButtonHover from "../public/images/addButtonHover.svg";
+import Image from "next/image";
+import { useState } from "react";
 
-export default function Task({ task, handleToggleTask }) {
+export default function Task({ task, handleToggleTask, setPage }) {
   const tasks = task.filter((item) => !item.checked);
+  const [hover, setHover] = useState(false);
```
Comment (F-Kirchhoff) [inline components/Task.js]: Is there now way that you can create the hove effect in CSS?

File: components/Task.js [hunk 7]
```diff
@@ -20,24 +25,51 @@ export default function Task({ task, handleToggleTask }) {
           />
         ))}
       </ul>
+      <AddButton
+        onMouseOver={() => setHover(!hover)}
+        onMouseOut={() => setHover(!hover)}
```
Comment (F-Kirchhoff) [inline components/Task.js]: Even though this is a quite verbose solution, the code is good!

File: components/Task.js [hunk 8]
```diff
@@ -1,9 +1,14 @@
 import React from "react";
```
Comment (F-Kirchhoff) [inline components/Task.js]: The naming of this component is a bit confusing, since it is not a single task that is displayed, but rather the whole tasklist

File: pages/index.js [hunk 9]
```diff
@@ -71,19 +73,50 @@ export default function Home() {
...
         <TaskCompleted task={task} handleToggleTask={handleToggleTask} />
-        <Task task={task} handleToggleTask={handleToggleTask} />
+        <Task
+          task={task}
+          handleToggleTask={handleToggleTask}
+          setPage={setPage}
+        />
+      </>
+    );
+  }
```
Comment (F-Kirchhoff) [inline pages/index.js]: is this a component inside another component call Home? If so you need to change this layout, you can remove "homepage and set the return value of this inner function as the return value of the outer function.

File: pages/index.js [hunk 10]
```diff
@@ -71,19 +73,50 @@ export default function Home() {
...
+
+  return (
+    <Wrap>
+      <WrapMask>
+        {page === "home" && <HomePage />}
+        {page === "create" && (
+          <Create appendTask={appendTask} setPage={setPage} />
+        )}
       </WrapMask>
```
Comment (F-Kirchhoff) [inline pages/index.js]: Why don't you use the default way of implementing routing in next.js? If you struggle sharing the state between two sites, have a look into Angelas latest code or ask her.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Mengzhen, A lot of your code is already ready to go, only how you established the page routing needs to be changed. Write a how to into the Github Discussions or ask Angela who had the same problem.

File: pages/index.js (hunk 10 above)
Comment (Mengzhen1992) [reply pages/index.js]: @F-Kirchhoff thanks for this suggestion! already finished!

File: pages/_app.js [hunk 11]
```diff
@@ -1,10 +1,71 @@
...
+    checked: false,
+  },
+];
+
+function generateID() {
+  return Math.random().toString(36).substring(2);
+}
 
 function MyApp({ Component, pageProps }) {
+  const [task, setTask] = useState(initialTasks);
```
Comment (Roland-Hufnagel) [inline pages/_app.js]: since initialTasks is an array with a lot of tasks, it is a good practice to give the useState the name 'tasks' and 'setTasks'

File: pages/index.js [hunk 12]
```diff
@@ -64,38 +25,20 @@ export default function Home() {
   }
 
   function handleToggleTask(id) {
-    const updatedTaskList = task.map((item) => {
+    const updatedTaskList = items.map((item) => {
```
Comment (Roland-Hufnagel) [inline pages/index.js]: why did you change from the name task to items? This makes me a little bit confused, because now your useState has the name 'item' and your setter-function has the name 'setTask'.

File: pages/create.js [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+    const formData = new FormData(event.target);
+    const { title, date, startTime, durationHour, durationMinute } =
+      Object.fromEntries(formData);
+    appendTask(title, durationHour, durationMinute);
+    router.push("/");
+  }
+  return (
+    <LayoutSytle>
+      <FormWrapMask>
+        <FormContainer>
```
Comment (Roland-Hufnagel) [inline pages/create.js]: Now you have a main in a div in a div in a div. It is a lot of wrapping here. Is that really necessary?

File: .vscode/launch.json [hunk 14]
```diff
@@ -0,0 +1,15 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
```
Comment (Roland-Hufnagel) [inline .vscode/launch.json:5]: What is this?

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]

**Review by Roland-Hufnagel — APPROVED** [review]
Looks very good now
